blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -276,14 +274,33 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
 
 object BlazeClientBuilder {
 
+  /**
+    * Indicates how to resolve SSLContext.
+    *  * NoSSL                = do not use SSL/HTTPS
+    *  * TryDefaultSSLContext = `SSLContext.getDefault()`, or `None` on systems where the default is unavailable
+    *  * Provided             = use the explicitly passed SSLContext
+    */
+  sealed trait SSLContextOption extends Product with Serializable
```

Comment:
I think this type can be `private[BlazeClientBuilder]`, or moved to `org.http4s.internal` if we apply this treatment to other backends.

Reply:
Hi @rossabaker - please see https://github.com/http4s/http4s/pull/3422/commits/63622ab6fe59d43ea58bda2d1440809dc2000232, namely its message. Let me know, please, if you recommend another approach.

Reply:
I think it could probably be straight up `private`, or `private[BlazeClientBuilder]`, but if the compiler disagrees, listen to it.

Reply:
> I think this type can be private[BlazeClientBuilder], or moved to org.http4s.internal if we apply this treatment to other backends.

I moved `SSLContextOption` to `org.http4s.internals`.

Please take a look at https://github.com/http4s/http4s/pull/3422#commits-pushed-8643f36, @rossabaker @hamnis.

Thanks.

> but if the compiler disagrees, listen to it.

Of course! :-)